src/__tests__/TimelineComponent.test.tsx

```diff
@@ -198,6 +199,204 @@ describe('<TimelineComponent /> R2 Tests\n', () => {
+    new Column('string', 1, 'Item Type', 1),
+    new Column('string', 1, 'Sales Channel', 1),
+    new Column('string', 1, 'Order Priority', 1),
+    new Column('date', 2, 'Order Date', 1),
+    new Column('number', 2, 'Order ID', 1),
+    new Column('date', 2, 'Ship Date', 1),
+    new Column('number', 2, 'Units Sold', 1),
+    new Column('number', 2, 'Unit Price', 1),
+    new Column('number', 2, 'Unit Cost', 1),
+    new Column('number', 2, 'Total Revenue', 1),
+    new Column('number', 2, 'Total Cost', 1),
+    new Column('number', 2, 'Total Profit', 1),
+    new Column('number', 2, 'index', 1),
+    new Column('number', 2, 'Order Date_num', 1)]);
+
+  // helper function to make innerHTML more readable
+  const prettyHTML = (htmlString: string) => {
+    const lines: Array<string> = htmlString.split(/(?=<)/g);
+    let tempString: string = '';
+    for (let i: number = 0; i < lines.length; i++) {
+      tempString += (lines[i] + '\n');
+    }
+    return tempString;
+  };
+
+  let wrapper: any;
+
+  beforeEach(() => {
+    // need to do this so that d3 works properly with jest/enzyme
+    wrapper = mount(
+        <TimelineComponent
+          data={data}
+        />);
+    document.body.innerHTML = wrapper.html();
+    wrapper = mount(
+        <TimelineComponent
+          data={data}
+        />);
+    // initialize & draw the timeline (updates html)
+    wrapper.instance().resetTimeline();
+
+    // Check that rendered TimelineComponent HTML matches snapshot
+    // This doesn't include the html created by d3
+    expect(wrapper.debug()).toMatchSnapshot();
+
+    // check d3 svgs match the snapshot
+    const svgTarget: any = document.getElementById('svgtarget');
+    let svgHTML: string = '';
+    if (svgTarget) {
+      svgHTML = prettyHTML(svgTarget.innerHTML);
+    }
+    // Check that svg created by d3 matches snapshot
+    expect(svgHTML).toMatchSnapshot();
+  });
+  // zoom in events for keydown & keyup
+  const zoomInEventDown = new KeyboardEvent('keydown', {'key': '+'});
+  const zoomInEventUp = new KeyboardEvent('keyup', {'key': '+'});
+  // zoom out events for keydown & keyup
+  const zoomOutEventDown = new KeyboardEvent('keydown', {'key': '-'});
+  const zoomOutEventUp = new KeyboardEvent('keyup', {'key': '-'});
+
+
+  it('T3.3 Snapshot Test\n', () => {
+    expect(wrapper.instance().getScale()).toBe(1.0);
+
+    // check scale for zoomInEventDown and zoomInEventUp separately
+    document.body.dispatchEvent(zoomInEventDown);
+    document.body.dispatchEvent(zoomInEventUp);
+    const tempScaleEventDown = wrapper.instance().getScale();
+    expect(tempScaleEventDown).toBeGreaterThan(1.0);
+
+    // find d3 svg element
+    const svgTarget: any = document.getElementById('svgtarget');
+    let svgHTML: string = '';
+    if (svgTarget) {
+      svgHTML = prettyHTML(svgTarget.innerHTML);
+    }
+
+    // Check that svg created by d3 matches snapshot
+    expect(svgHTML).toMatchSnapshot();
```

Comment:
Why does this work? If the zoom in was done, it should have been a different display so it shouldn't match the original snapshot.